File: Remove Duplicates from Sorted Array II.c
#include <stdio.h>

int removeDuplicates(int *array,int size){
	if(size<=1)
		return size;
	int prev=*array,i;
	int nextpos=1,counter=1;
	for(i=1;i<size;++i)
		if(*(array+i)!=prev){
			prev=*(array+i);
			counter=1;
			*(array+nextpos++)=*(array+i);
		}else if(counter<2){
			*(array+nextpos++)=*(array+i);
			counter++;
		}
	return nextpos;
}

int main(){
	int array[]={1,1,1,2,2,3},size=sizeof array/sizeof *array;
	int newsize;
	printf("%d\n",newsize=removeDuplicates(array,size));
	for(int i=0;i<newsize;++i)
		printf("%d ",array[i]);
	printf("\n");
}

File: Remove Duplicates from Sorted Array II.py
class Solution:
    def removeDuplicates(self,array):
        if not array:
            return 0
        addPos = 1
        counter = 1
        prev=array[0]
        for num in array[1:]:
            if num==prev:
                counter+=1
                if counter<=2:
                    array[addPos]=num
                    addPos+=1
            else:
                counter=1
                prev=num
                array[addPos]=num
                addPos+=1
        return addPos

ins=Solution()

print ins.removeDuplicates([1,1,1,2,2,3])